### PR TITLE
added spinner after training button

### LINKED ISSUE #861: [FEATURE]: Spinner after clicking Train button
### Feature Name

Spinner after clicking Train button

### Your Name

Faris Durrani

### Description

After clicking on Train to train a dataset, there is a couple of seconds of lag where the screen seems frozen but the app is querying the backend. We want better UI feedback and we can simply do this by:
- adding a simple spinner [example](https://mui.com/material-ui/react-progress/) after the user clicks on Train
- disabling Train button while it's loading

### PR DESCRIPTION
**Pull request by prishavall**
# Added Spinner after Training Button

**What user problem are we solving?**
Needed a spinner icon that shows that the page is loading. The spinner also needs to be unclickable. After clicking the button to where the spinner shows up the train button needs to be disabled.
**What solution does this PR provide?**
Code added for inserting the spinner icon and disabling the train button with tabular/image
**Testing Methodology**
How did you test your changes and verify that existing 
functionality is not broken
Tested on localhost:3000 both image/tabular
 [New Recording - 7/14/2023, 1:36:14 PM](https://vimeo.com/845362166/8e3e9edf59)
**Any other considerations**

### CODE REVIEW

File: frontend/src/features/Train/features/Image/components/ImageTrainspace.tsx [hunk 1]
```diff
@@ -10,6 +10,7 @@ import {
   StepButton,
```
Comment (karkir0003) [inline frontend/src/features/Train/features/Image/components/ImageTrainspace.tsx:10]: Can you add a screen recording of your feature in action for Image Trainspace and Tabular Trainspace? 

@prishavall

File: frontend/src/features/Train/features/Tabular/components/TabularTrainspace.tsx [hunk 2]
```diff
@@ -91,6 +92,8 @@ const TrainspaceStepInner = ({
 }) => {
   const Component = STEP_SETTINGS[TRAINSPACE_SETTINGS.steps[step]].component;
   const [isStepModified, setIsStepModified] = useState<boolean>(false);
+  //create a state to track is the button clicked?
```
Comment (karkir0003) [inline frontend/src/features/Train/features/Tabular/components/TabularTrainspace.tsx]: remove this comment

**Review by karkir0003 — COMMENTED** [review]
@prishavall looks good to me. Left some minor comments to address. Other than that, I conditionally approve this PR based on @farisdurrani 's review

**Comment by karkir0003** [comment]
Also, please fill in the "Title of PR field" at the top of the PR description @prishavall

File: frontend/src/features/Train/features/Image/components/ImageTrainspace.tsx (hunk 1 above)
Comment (prishavall) [reply frontend/src/features/Train/features/Image/components/ImageTrainspace.tsx:10]: [New Recording - 7/14/2023, 1:36:14 PM](https://vimeo.com/845362166/8e3e9edf59)

Comment (karkir0003) [reply frontend/src/features/Train/features/Image/components/ImageTrainspace.tsx:10]: can you add this link to the pr description

@prishavall

File: frontend/src/features/Train/features/Tabular/components/TabularTrainspace.tsx (hunk 2 above)
Comment (karkir0003) [reply frontend/src/features/Train/features/Tabular/components/TabularTrainspace.tsx]: @prishavall

**Comment by farisdurrani** [comment]
This is very good @prishavall . For aesthetic reasons, can you try putting the spinner inside the TRAIN button instead so it doesn't create an awkward layout

**Review by farisdurrani — APPROVED** [review]

**Comment by karkir0003** [comment]
> This is very good @prishavall . For aesthetic reasons, can you try putting the spinner inside the TRAIN button instead so it doesn't create an awkward layout

could you put this as a suggestion in github PR (so that you can show what the code would look like with your suggestion) @farisdurrani

**Comment by farisdurrani** [comment]
> > This is very good @prishavall . For aesthetic reasons, can you try putting the spinner inside the TRAIN button instead so it doesn't create an awkward layout
> 
> could you put this as a suggestion in github PR (so that you can show what the code would look like with your suggestion) @farisdurrani

I would rather not. The change would be too big for a code suggestion, and the code implementation here is less important than the aesthetic goal

**Pull request merged**